src/services/vcmap.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  CesiumMap,
+  CesiumTilesetLayer,
+  LayerCollection,
+  MapCollection,
+  OpenlayersMap,
+  OpenStreetMapLayer,
+  TerrainLayer,
+  VcsMap,
+  VectorTileLayer,
+  Viewpoint,
+} from '@vcmap/core'
+
+const cleanUpMap = (): void => {
+  if (window.mapContext) {
+    window.mapContext.destroy()
+    window.mapContext = null
+  }
+}
+
+const initMapCollection = (target: string, map: VcsMap): MapCollection => {
+  const mapCollection = new MapCollection()
+
+  mapCollection.setTarget(target)
+  mapCollection.add(map)
+
+  return mapCollection
+}
+
+const addLayers = async (layers: LayerCollection) => {
+  const osm = new OpenStreetMapLayer({ name: 'osmBase' })
+  layers.add(osm)
+
+  // Adding terrain
+  layers.add(
+    new TerrainLayer({
+      name: 'terrain',
+      url: 'data/terrain',
+    })
+  )
+
+  layers.add(
+    new CesiumTilesetLayer({
+      name: 'building',
+      url: 'data/buildings',
+    })
+  )
+
+  // Adding VectorTiles
+  const vectorTiles = new VectorTileLayer({
+    mapNames: ['ol'],
+    tileProvider: {
+      type: 'MVTTileProvider',
+      url: 'data/vectorTiles/{z}/{x}/{y}.pbf',
+      baseLevels: [12, 13, 14, 15],
+      idProperty: 'gmlid',
+    },
+    extent: {
+      coordinates: [8.6521196, 50.7494957, 8.8483077, 50.8534806],
+      epsg: 4326,
+    },
+  })
+  layers.add(vectorTiles)
+
+  Promise.all(
+    layers._array.map(async (layer) => {
+      layer.activate()
+    })
+  )
+}
+
+const clearTargetElement = (target: string) => {
+  const targetElem: HTMLElement | null = document.getElementById(target)
+  if (targetElem) {
+    while (targetElem.firstChild) {
+      targetElem.removeChild(targetElem.firstChild)
+    }
+  }
+}
+
+export default async function initMap(
```

Comment:
I dont understand the divergences with https://github.com/virtualcitySYSTEMS/map-core-demo/blob/dbde70bdefa5e6ae252a8545ec9a7068e5ef8d50/src/api/api.js#L27

Could explain why you haven't just copied this file ?